src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -539,32 +542,33 @@ public static void TestRoundtrippingSpecificCodepageEncoding(string encodingName
         [MemberData(nameof(CodePageInfo))]
         public static void TestCodepageEncoding(int codePage, string webName, string queryString)
         {
-            Encoding encoding;
-            // There are two names that have duplicate associated CodePages. For those two names,
-            // we have to test with the expectation that querying the name will always return the
-            // same codepage.
-            if (codePage != 20932 && codePage != 50222)
-            {
-                encoding = CodePagesEncodingProvider.Instance.GetEncoding(queryString);
-                Assert.Equal(encoding, CodePagesEncodingProvider.Instance.GetEncoding(codePage));
-                Assert.Equal(encoding, CodePagesEncodingProvider.Instance.GetEncoding(webName));
-            }
-            else
-            {
-                encoding = CodePagesEncodingProvider.Instance.GetEncoding(codePage);
-                Assert.NotEqual(encoding, CodePagesEncodingProvider.Instance.GetEncoding(queryString));
-                Assert.NotEqual(encoding, CodePagesEncodingProvider.Instance.GetEncoding(webName));
-            }
+            Encoding encoding = VerifyCodepageEncoding(codePage, webName, queryString);
 
-            Assert.NotNull(encoding);
-            Assert.Equal(codePage, encoding.CodePage);
-            Assert.Equal(webName, encoding.WebName);
+            // Small round-trip for ASCII alphanumeric range (some code pages use different punctuation!)
+            // Start with space.
+            char[] traveled = encoding.GetChars(encoding.GetBytes(s_asciiPrintable));
+            Assert.Equal(s_asciiPrintableCharArr, traveled);
+        }
+
+        [Theory]
+        [MemberData(nameof(CodePageInfo))]
+        public static void TestEncoderNLSAndDecoderNLS(int codePage, string webName, string queryString)
+        {
+            Encoding encoding = VerifyCodepageEncoding(codePage, webName, queryString);
```

Comment:
>Encoding encoding = VerifyCodepageEncoding(codePage, webName, queryString); [](start = 12, length = 75)

This is already validated in TestCodepageEncoding, why we validate it here again?

Reply:
We shouldn't. I've removed the duplication by leaving the old code unchanged, and extracting only code needed for creating the encoding (without the assertions) in the new test.